infer/tests/check_test.go

```diff
@@ -257,3 +259,45 @@ func TestCheckDefaultsRecursive(t *testing.T) {
 	}, resp.Inputs)
 
 }
+
+func TestCheckAlwaysAppliesSecrets(t *testing.T) {
```

Comment:
Maybe it's just me, but I feel like this test would benefit from a high-level description of how the parts fit together. Like how the secret-ness of the input comes in via the annotation on `CustomCheckNoDefaultsArgs.Input`.